### PR TITLE
Milestone 5 : Testing

### PR DESCRIPTION
**Pull request by MarwanDev**
### In this milestone we:

- Wrote tests for operate.js and calculate.js using jest
- Created snapshots for Home.jsx and calculator.jsx
- Wrote tests for Home.jsx, Navbar.jsx, Quotes.jsx and calculator.jsx using react-testing library

### CODE REVIEW

File: src/tests/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      operation: 'x',
+    };
+    const output = calculate(btn, 'AC');
+    expect(output).toEqual({
+      total: null,
+      next: null,
+      operation: null,
+    });
+  });
+});
```
Comment (TedLivist) [inline src/tests/calculate.test.js:31]: - [ ] Kindly add more tests covering all the functionalities and scenarios in the `calculate.js` file

File: src/tests/calculator.test.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import renderer from 'react-test-renderer';
+import { Calculator } from '../components';
+
+it('renders calculator page correctly', () => {
+  const tree = renderer
+    .create(<Calculator />)
+    .toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (TedLivist) [inline src/tests/calculator.test.js:9]: - [ ] You are missing tests for simulating user interactions. Kindly add user interaction tests using the React Testing Library.

  If you are unfamiliar with it you can visit [this tutorial](https://www.robinwieruch.de/react-testing-library) on how to add tests for that.

File: src/tests/__snapshots__/home.test.js.snap [hunk 3]
```diff
@@ -0,0 +1,47 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
```
Comment (TedLivist) [inline src/tests/__snapshots__/home.test.js.snap:2]: - [ ] You only have snapshot tests for the Calculator and Home components, but you are required to add snapshot tests for all your component. Kindly add snapshot tests for the Layout, Navbar, and Quotes components.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @MarwanDev and @rushemaa,

Good job so far!
✔️ Well-written and passing tests
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @MarwanDev and @rushemaa,

## APPROVED 🟢

## Highlights
✔️ Passing tests

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**